src/app/3d/qgs3dmaptoolpointcloudchangeattribute.cpp

```diff
@@ -284,10 +323,13 @@ QVector<int> Qgs3DMapToolPointCloudChangeAttribute::selectedPointsInNode( const
     pt.setX( ptScreen.x() );
     pt.setY( ptScreen.y() );
 
-    if ( searchPolygon.intersects( &pt ) )
-    {
-      selected.append( i );
-    }
+    if ( !searchPolygon.intersects( &pt ) )
+      continue;
+
+    if ( pointIsClipped( mapToPixel3D.origin, clipPlanes, x, y, z ) )
```

Comment:
Wouldn't it make sense to skip the method call completely if clipPlanes.isEmpty() ?
This is going to be called _a lot!_